tests/test_barostat.py

```diff
@@ -167,15 +168,22 @@ def test_barostat_varying_pressure():
     baro.set_pressure((1 * unit.atmosphere).value_in_unit(unit.bar))
     # Changing the barostat interval resets the barostat step.
     baro.set_interval(2)
-    start = time.time()
-    volumes = []
-    while time.time() - start < 15:
-        ctxt.multiple_steps(np.ones(1000)*lam)
-        atm_box = ctxt.get_box()
-        volumes.append(compute_box_volume(atm_box))
 
+    ctxt.multiple_steps(np.ones(5000)*lam)
+    atm_box = ctxt.get_box()
+    platform_version = platform.version().lower()
+    # OpenEye's AM1 Charging values are OS platform dependent. To ensure that we have deterministic values
+    # we check against our two most common OS versions, Ubuntu 18.04 and 20.04.
+    box_vol = 7.594815077349339
+    box_diff = 0.26980723017633856
+    if "ubuntu" not in platform_version:
+        print("Test expected to run under ubuntu 20.04 or 18.04")
+    if "20.04" in platform_version:
+        box_vol = 7.838565243228307
+        box_diff = 0.5460385300302901
```

Comment:
Request: Could we add an environment-dependent assertion about starting conditions as well? (Starting conditions (namely, the partial charge parameters), but not any of the later simulation logic, are expected to be environment-dependent. This should imply currently tested values, I think, but asserting both would make it faster to track down whether the OS-dependent starting condition assumption or the OS-independent simulation logic assumption change in the future.)

Reply:
Thanks! Resolved in https://github.com/proteneer/timemachine/pull/458/commits/413f8770c1f88521658c9b5ab2b4cd34f6ac8b7a